libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -93,6 +101,41 @@ class AudioSource(TypedDict):
     type: str
 
 
+class FileSource(TypedDict):
+    src: str
+    type: str
+
+
+def create_csv_file(
+    dataset: str,
+    config: str,
+    split: str,
+    data: Any,
+    headers: List[str],
+    assets_base_url: str,
+    file_name: str,
+    assets_directory: StrPath,
+) -> FileSource:
+    dir_path, url_dir_path = create_file_dir(
+        dataset=dataset,
+        config=config,
+        split=split,
+        assets_directory=assets_directory,
+    )
+    makedirs(dir_path, ASSET_DIR_MODE, exist_ok=True)
+    file_path = dir_path / file_name
+
+    with open(file_path, "w") as output_file:
+        fc = csv.DictWriter(
+            output_file,
+            fieldnames=headers,
+        )
+        fc.writeheader()
+        fc.writerows(data)
+
+    return {"src": f"{assets_base_url}/{url_dir_path}/{file_name}", "type": "csv"}
```

Comment:
```suggestion
    return {"src": f"{assets_base_url}/{url_dir_path}/{file_name}", "type": "text/csv"}
```

https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types